src/js/view/floating/floating-controller.ts

```diff
@@ -200,6 +212,7 @@ export default class FloatingController {
             return;
         }
 
+        preTransition(this);
```

Comment:
Shouldn't `preTransition` be called right before `removeClass(this._playerEl, 'jw-flag-floating');` ? Removing and adding classes at once rather than at different intervals (with the timeout below) should improve performance, and keeping the lines of style changes together reduces complexity.

Reply:
Has to also cover mobileFloatIntoPlace